pkg/api/types.go

```diff
@@ -2071,13 +2094,14 @@ func (p *Properties) GetKubernetesHyperkubeSpec() string {
 
 // IsAzureStackCloud return true if the cloud is AzureStack
 func (p *Properties) IsAzureStackCloud() bool {
-	return p.CustomCloudProfile != nil
+	// For backward compatibility, treat nil Environment and empty Environment name as AzureStackCloud as well
+	return p.CustomCloudProfile != nil && (p.CustomCloudProfile.Environment == nil || p.CustomCloudProfile.Environment.Name == "" || strings.EqualFold(p.CustomCloudProfile.Environment.Name, "AzureStackCloud"))
```

Comment:
nit:

return p.IsCustomCloudProfile() instead of copying the `p.CustomCloudProfile != nil` implementation from the `IsCustomCloudProfile` method (i.e., let's just re-use the method)

@jadarsie do we want the `p.CustomCloudProfile.Environment == nil` back-compat foo for AzureStackCloud? It would be nice to have `IsAzureStackCloud` actually require that the `Environment.Name` string property be equal to `"AzureStackCloud"`.